### PR TITLE
Alvin/dbus

### PR DESCRIPTION
**Pull request by alvinsunyixiao**
DBUS implemented + Migrate to HAL v1.25.0

I apologize for squishing those two changes in one review. My updated CubeMX is not working properly with the old version HAL libraries so I just updated it all together. Please take a look and test out some examples and see if anything is broken.

### CODE REVIEW

**Comment by AlchemicRonin** [comment]
Do I need to test the new code on real machines? :)

**Comment by RogerQi** [comment]
It'd always be desirable to test code on a physical machine. But to test DBUS you would probably need a controller.

File: examples/buzzer/main.cc [hunk 1]
```diff
@@ -30,7 +30,7 @@ static bsp::BuzzerNoteDelayed Mario[] = {
     {Note::Silent, 240}, {Note::Mi3M, 80},   {Note::Silent, 240},
     {Note::Do1M, 80},    {Note::Silent, 80}, {Note::Mi3M, 80},
     {Note::Silent, 240}, {Note::So5M, 80},   {Note::Silent, 560},
-    {Note::So5L, 80},
+    {Note::So5L, 80},    {Note::Silent, 0},  {Note::Finish, 0}
```
Comment (AlchemicRonin) [inline examples/buzzer/main.cc:33]: The test on board shows that Finish is necessary or the next song will not play.

File: examples/buzzer/main.cc [hunk 2]
```diff
@@ -62,7 +62,7 @@ static bsp::BuzzerNoteDelayed War_Cant_of_Mars[] = {
     //    {Note::Re2H, 200},   {Note::Mi3H, 1600},  {Note::Do1H, 800},
     //    {Note::Do1H, 800},   {Note::Re2H, 800},   {Note::Re2H, 800},
     //    {Note::Do1H, 800},   {Note::Si7M, 800},   {Note::Do1H, 1600},
-    //    {Note::Silent, 800}, {Note::Silent, 800}, {Note::Finish, 0},
+    {Note::Silent, 0}, {Note::Finish, 0},
```
Comment (AlchemicRonin) [inline examples/buzzer/main.cc:65]: Without Silent, buzzer will not stop PWM signal.

File: examples/buzzer/main.cc (hunk 1 above)
Comment (alvinsunyixiao) [reply examples/buzzer/main.cc:33]: Let's put a separate fix on those things as they are unrelated to this pull request.

**Pull request merged**